src/test/scala/fte/data/GraphCreator.scala

```diff
@@ -0,0 +1,140 @@
+ along with this program. If not, see http://www.gnu.org/licenses/gpl.html dibl
+*/
+package fte.data
+
+import dibl.Face.facesFrom
+import dibl.LinkProps.WhiteStart
+import dibl.proto.TilesConfig
+import dibl.{ Diagram, LinkProps, NewPairDiagram, NodeProps, SimpleLink, ThreadDiagram }
+import fte.layout.OneFormTorus
+
+import scala.collection.JavaConverters._
+
+object GraphCreator {
+
+  /**
+    * @param urlQuery parameters for: https://d-bl.github.io/GroundForge/tiles?
+    *                 For now the tile must be a checker tile and
+    *                 the patch size must span 3 columns and 2 rows of checker tiles.
+    *                 A simplified ascii-art view of a pattern definition:
+    *                 +----+----+----+
+    *                 |....|....|....|
+    *                 |....|....|....|
+    *                 +----+----+----+
+    *                 |....|XXXX|....|
+    *                 |....|XXXX|....|
+    *                 +----+----+----+
+    * @return The X's in the pattern definition are added to the returned graph.
+    */
+  def fromDiagram(urlQuery: String): Option[Graph] = {
+    implicit val config: TilesConfig = TilesConfig(urlQuery)
+
+    implicit val diagram: Diagram = ThreadDiagram(NewPairDiagram.create(config))
+    implicit val scale: Int = 2
+    //    implicit val diagram: Diagram = NewPairDiagram.create(config)
+    //    implicit val scale: Int = 1
+
+    val links = diagram.links.filter(inCenterBottomTile)
+      .sortBy(l => s"${ diagram.node(l.target).id }${ diagram.node(l.source).id }")
+    val graph = new Graph()
+
+    // create each vertex on the torus once
+    val vertexMap = links.map(_.target).distinct.zipWithIndex
+      .map { case (nodeNr, i) =>
+        // The initial coordinates don't matter as long as they are unique.
+        diagram.node(nodeNr).id -> graph.createVertex(i, 0)
+      }.toMap
+    println(vertexMap.keys.toArray.sortBy(identity).mkString(","))
+
+    // create edges of one tile
+    links.foreach { link =>
+      val source = diagram.node(link.source)
+      val target = diagram.node(link.target)
+      val whiteStart = link.isInstanceOf[WhiteStart]
+      val (dx, dy) = deltas(whiteStart, source, target)
+      println(s"(${ source.id },${ target.id }) deltas($dx,$dy) ${ source.isLeftTwist }, ${ source.isRightTwist }, ${ target.isLeftTwist }, ${ target.isRightTwist }, $whiteStart")
+      graph.createEdge(vertexMap(source.id), vertexMap(target.id), dx, dy)
+    }
+
+    def faces = {
+      val edges = graph.getEdges.asScala.toArray
+      def findEdge(link: SimpleLink): Option[Edge] = edges.find(edge =>
+        edge.getStart == vertexMap(link.sourceId) &&
+          edge.getEnd == vertexMap(link.targetId)
+      )
+      facesFrom(links)
+        .map(scalaFace => new Face() {
+          println(scalaFace)
+          setEdges(new java.util.LinkedList[Edge]() {
+            (scalaFace.leftArc ++ scalaFace.rightArc)// TODO what order should the edges get?
```

Comment:
the next action to take together with
https://github.com/d-bl/GroundForge/blob/2b35fd2eb9b8973f70fd38b1b3353c8c590ddd42/src/test/scala/fte/layout/OneFormTorus.java#L18-L21